packages/office-ui-fabric-react/src/components/Layer/examples/Layer.Customized.Example.tsx

```diff
@@ -1,84 +1,52 @@
 import { Toggle } from 'office-ui-fabric-react/lib/Toggle';
-import { LayerHost } from 'office-ui-fabric-react/lib/Layer';
+import { LayerHost, ILayerProps } from 'office-ui-fabric-react/lib/Layer';
 import { Panel } from 'office-ui-fabric-react/lib/Panel';
-import * as React from 'react';
+import { IFocusTrapZoneProps } from 'office-ui-fabric-react/lib/FocusTrapZone';
+import { mergeStyles } from 'office-ui-fabric-react/lib/Styling';
+import { Customizer } from 'office-ui-fabric-react/lib/Utilities';
+import { useId, useToggle, useBoolean } from '@uifabric/react-hooks';
 
-export interface ILayerCustomizedExampleState {
-  showPanel: boolean;
-  trapPanel: boolean;
-}
+const layerHostClass = mergeStyles({
+  position: 'relative',
+  height: 400,
+  overflow: 'hidden',
+  border: '1px solid #ccc'
+});
 
-export class LayerCustomizedExample extends React.Component<{}, ILayerCustomizedExampleState> {
-  public state: ILayerCustomizedExampleState = {
-    showPanel: false,
-    trapPanel: false
-  };
-  // Use getId() to ensure that the ID is unique on the page.
-  // (It's also okay to use a plain string without getId() and manually ensure uniqueness.)
-  private _layerHostId: string = getId('layerHost');
+const focusTrapZoneProps: IFocusTrapZoneProps = {
+  isClickableOutsideFocusTrap: true,
+  forceFocusInsideTrap: false
+};
 
-  public render(): JSX.Element {
-    return (
-      <div>
-        <p>
-          A <code>Panel</code> is rendered, trapped in a specified container. Use <b>Show panel</b> to show/hide the panel (or click the X
-          to dismiss it). Use <b>Trap panel</b> to release the panel from its bounds.
-        </p>
-        <Toggle label="Show panel" inlineLabel checked={this.state.showPanel} onChange={this._onShowPanelChange} />
-        <Toggle label="Trap panel" inlineLabel checked={this.state.trapPanel} onChange={this._onTrapPanelChange} />
-        <Customizer
-          scopedSettings={
-            this.state.trapPanel
-              ? {
-                  Layer: {
-                    hostId: this._layerHostId
-                  }
-                }
-              : {}
-          }
-        >
-          {this.state.showPanel && (
-            <Panel
-              isOpen={true}
-              hasCloseButton={true}
-              headerText="Test"
-              focusTrapZoneProps={{
-                isClickableOutsideFocusTrap: true,
-                forceFocusInsideTrap: false
-              }}
-              onDismiss={this._onDismissPanel}
-            />
-          )}
-        </Customizer>
-        <LayerHost
-          id={this._layerHostId}
-          style={{
-            position: 'relative',
-            height: '400px',
-            overflow: 'hidden',
-            border: '1px solid #ccc'
-          }}
-        />
-      </div>
-    );
-  }
+export const LayerCustomizedExample: React.FunctionComponent = () => {
+  const [isPanelOpen, showPanel, dismissPanel] = useBoolean(false);
```

Comment:
Hey Elizabeth; one confusion I've had here that I think is a problem:

Typically `useState` returns a double; an array consisting of the value and a function to set it.

When we have complex hooks returning multiple things, arrays are a terrible way to return values. There is no predictability, they can't scale we can't rename or repurpose anything. I think this situation here illustrates is. `useBoolean` is meaningless to me without digging into the code. It returns 3 values, rather than 2. It seems that the second and third items are setToTrue and setToFalse functions. I might also need a toggle function. 

(And by the way I had the same question in my head on `useToggle` earlier; it returns a toggle function, but what about setToTrue and setToFalse?)

I think we need a pattern here; if you are returning non standard things, I think it should be in an object. That is:

```
const { 
  isTrue: isPanelOpen,
  setTrue: showPanel,
  setFalse: dismissPanel,
  toggle: togglePanel
} = useBoolean(false);
```
Now all of these are optional, you can still rename them, but they are far more explicit and type safe.

Thoughts?

Reply:
I agree, more than 2 starts to get messy. From you example, I didn't realize there was a 4th. 

Here's an example in the wild

https://streamich.github.io/react-use/?path=/story/state-usequeue--docs

```js
const { add, remove, first, last, size } = useQueue();
```

Reply:
The other option i've seen is to return the actions grouped as the 2nd array item

```js
const [isTrue, actions] = useBoolean(false);
// or
const [isTrue, {setTrue, setFalse, toggle}] = useBoolean(false);
```

like this example 

https://streamich.github.io/react-use/?path=/story/state-useupsert--docs
```js
const [list, { set, upsert, remove }] = useUpsert(comparisonFunction, initialItems);
```

Reply:
useBoolean looks more like useUpsert where you have 1 value return and one set of actions. I'd suggest that one.

Reply:
I like removing `useToggle` and having `useBoolean` return setters for true/false/toggle. Not sure whether I like the object syntax or array syntax...the object syntax is more verbose but less error-prone than full or partial array syntax.

Reply:
I just updated to have only useBoolean and entirely use the object approach. Curious what you both think.